### PR TITLE
master-port 49822 and 52217

### PR DESCRIPTION
**Pull request by mchugh19**
### What does this PR do?
Ports #49822 and #52217 to master branch

### Tests written?
Yes - as originally included

Note: this includes the release notes from Neon as of when this was committed. Might be a conflict'y item in this process.

### CODE REVIEW

File: doc/topics/releases/neon.rst [hunk 1]
```diff
@@ -3,3 +3,290 @@
...
+
+    /srv/salt/apache/
+        init.sls
+        config.sls
+        saltcheck-tests/
+            init.tst
+            config.tst
+            deployment_validation.tst
+
+Tests can be run for each state by name, for all apache/saltcheck/*.tst files, or for all states
```
Comment (waynew) [inline doc/topics/releases/neon.rst]:
```suggestion
Tests can be run for each state by name, for all ``apache/saltcheck/*.tst files``, or for all states
```

File: salt/modules/xml.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+'''
+XML file mangler
```
Comment (waynew) [inline salt/modules/xml.py]: mangler or manager? Is it supposed to rip, shred, and tear xml files (Who doesn't want that? :trollface: )

File: salt/states/xml.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+'''
+XML Mangler
```
Comment (waynew) [inline salt/states/xml.py]: Probably meant `XML Manager` here, too?

**Review by waynew — COMMENTED** [review]
Thanks for the PR! Code looks great - a couple of questions about the comments, though.

File: doc/topics/releases/neon.rst [hunk 4]
```diff
@@ -3,3 +3,290 @@
...
+- The :py:func:`file.rename <salt.states.file.rename>` state will now return a
+  ``True`` result (and make no changes) when the destination file already
+  exists, and ``Force`` is not set to ``True``. In previous releases, a
+  ``False`` result would be returned, but this meant that subsequent runs of
+  the state would fail due to the destination file being present.
+
+- The ``onchanges`` and ``prereq`` :ref:`requisites <requisites>` now behave
+  properly in test mode.
+
+Module Changes
```
Comment (dwoz) [inline doc/topics/releases/neon.rst]: @mchugh19 It looks like this is including some changes that are not in the original PRs. Can you please have a look?

**Review by dwoz — CHANGES_REQUESTED** [review]

File: salt/modules/xml.py (hunk 2 above)
Comment (mchugh19) [reply salt/modules/xml.py]: It was a working title :) Adjusted!

File: doc/topics/releases/neon.rst (hunk 4 above)
Comment (mchugh19) [reply doc/topics/releases/neon.rst]: Release notes corrected

File: doc/topics/releases/neon.rst (hunk 1 above)
Comment (mchugh19) [reply doc/topics/releases/neon.rst]: Corrected in the proper saltcheck PR :)

File: salt/states/xml.py (hunk 3 above)
Comment (mchugh19) [reply salt/states/xml.py]: Done!

**Review by waynew — COMMENTED** [review]
I think code-wise, this looks good now. Just a couple of tweaks and it looks ready for merge.

File: tests/unit/states/test_xml.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+'''
+# Import Python libs
+from __future__ import absolute_import, print_function, unicode_literals
+
+# Import Salt Testing Libs
+from tests.support.mixins import LoaderModuleMockMixin
+from tests.support.unit import skipIf, TestCase
+from tests.support.mock import (
+    NO_MOCK,
+    NO_MOCK_REASON,
```
Comment (waynew) [inline tests/unit/states/test_xml.py]: Ah - apparently that *other* change got merged in. There's no such complaint as `NO_MOCK` and `NO_MOCK_REASON` anymore. Any usages of these two in the PR should be removed.

**Review by waynew — CHANGES_REQUESTED** [review]
One final change(?)

File: tests/unit/states/test_xml.py (hunk 5 above)
Comment (mchugh19) [reply tests/unit/states/test_xml.py]: removed!

File: tests/unit/modules/test_xml.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        <foo:singers>
+            <foo:singer id="4">Tom Waits</foo:singer>
+            <foo:singer id="5">B.B. King</foo:singer>
+            <foo:singer id="6">Ray Charles</foo:singer>
+        </foo:singers>
+    </root>
+    '''
+
+
+@skipIf(NO_MOCK, NO_MOCK_REASON)
```
Comment (waynew) [inline tests/unit/modules/test_xml.py]: Doh - one more?

(also, perhaps fire off a `grep -Inr NO_MOCK tests/` to make sure we're not missing any others)

Comment (mchugh19) [reply tests/unit/modules/test_xml.py]: now no_mock clean

Comment (waynew) [reply tests/unit/modules/test_xml.py]: [so](https://jenkinsci.saltstack.com/job/pr-lint/job/PR-54977/10/pylint/folder.592290234/) [close](https://jenkinsci.saltstack.com/job/pr-lint/job/PR-54977/10/pylint/folder.10734287/) (the skipIf imports are now unused 😬 )

We'll get there! 😂

Comment (mchugh19) [reply tests/unit/modules/test_xml.py]: Things are slightly better today than they were yesterday ;)

**Comment by mchugh19** [comment]
test failures appear to be unrelated to this PR

**Pull request merged**